gammapy/spectrum/tests/test_spectrum_extraction.py

```diff
@@ -29,17 +57,58 @@ def test_spectrum_extraction(tmpdir):
 
     bounds = EnergyBounds.equal_log_spacing(1, 10, 40, unit='TeV')
 
-    obs = [23523, 23559, 11111]
+    obs = [23523, 23559, 11111, 23592]
     store = gammapy_extra.filename("datasets/hess-crab4-hd-hap-prod2")
     ds = DataStore.from_dir(store)
 
     ana = SpectrumExtraction(datastore=ds, obs_ids=obs, on_region=on_region,
-                           bkg_method=bkg_method, exclusion=excl,
-                           ebounds=bounds)
+                             bkg_method=bkg_method, exclusion=excl,
+                             ebounds=bounds)
 
-    #test methods on SpectrumObservationList
+    # test methods on SpectrumObservationList
     obs = ana.observations
-    assert len(obs) == 2
-    obs23523 = obs.get_obs_by_id(23523)
+    assert len(obs) == 3
+    obs23523 = obs.get_obslist_from_obsid([23523])[0]
     assert obs23523.on_vector.total_counts == 123
+    new_list = obs.get_obslist_from_obsid([23523, 23592])
+    assert new_list[0].obs_id == 23523
+    assert new_list[1].obs_id == 23592
+
+@requires_data('gammapy-extra')
+def test_spectrum_extraction_grouping_from_an_observation_list():
+    """
+    Test the stacking for two observations
+    """
+
+    ana = make_spectrum_extraction()
+    ana.extract_spectrum()
```

Comment:
Here I get
```
E   KeyError: u'MUONEFF'
```
I suppose you have not commited your changed eventlist to ``gammapy-extra``